sdk/storage/Azure.Storage.Common/tests/Shared/TestEventListener.cs

```diff
@@ -21,21 +22,38 @@ namespace Azure.Storage.Test
     /// </summary>
     internal class TestEventListener : AzureEventSourceListener
     {
+        private static StringBuilder s_eventBuffer;
```

Comment:
It feels weird to create instances but dump their output in a static buffer.  Why not make this an instance member?

Reply:
The reason for all of the statics is that we need to pass an Action to the base ctor in TestEventListener ctor. This forces the method used in the action to be a static.

Reply:
Should we instead be deriving from the TestEventListener class in Core.Testing? That class doesn't seal OnEventWritten so we could override it to do our logging.

Reply:
Don't inherit from  AzureEventSourceListener, instantiate it in your ctor.